tests/fixtures.py

```diff
@@ -211,6 +213,62 @@ def setUp(self):
         build_files(EggInfoPkg.files, prefix=self.site_dir)
 
 
+class EggInfoPkgPipInstalledNoModules(OnSysPath, SiteDir):
```

Comment:
Why is `NoModules` tested but not `WithModules` (presumably the more common use-case)? I'm particularly concerned that the "relative to egg-info directory" logic may prove problemmatic for package modules, especially since the egg-info directory isn't always in a reliable location relative to sources.

Reply:
Good catch, I'm adding a `EggInfoPkgPipInstalledNoToplevel` class now that does include a module that should be inferred via `installed-files.txt`. And there are indeed bugs/complications here. Will fix.

Reply:
Added `EggInfoPkgPipInstalledNoToplevel` in https://github.com/python/importlib_metadata/pull/437/commits/110f00d24304fdf446a640513f6bf7aef18ba9c3, and fixed associated issue in https://github.com/python/importlib_metadata/pull/437/commits/eeb2ed1593e25e4be3c7da8db073260a4ae8ec1e